argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+import glob
+import json
+import os
+import pickle
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.interpolate as interpolate
+
+import argoverse
+import open3d as o3d
+import torch
+import torch.nn.functional as F
+from argoverse.data_loading.argoverse_tracking_loader import ArgoverseTrackingLoader
+from argoverse.utils.json_utils import read_json_file
+
+dict_color: Dict[str, Tuple[float, float, float]] = {}
+dict_color["easy"] = (0.0, 1.0, 0.0)  # green
+dict_color["far"] = (0.0, 0.4, 0.0)  # dark green
+dict_color["occ"] = (0.0, 0.0, 1.0)  # red
+dict_color["fast"] = (0.0, 1.0, 1.0)  # yellow
+dict_color["short"] = (0.8, 0.3, 0.3)  # dark purple
+
+list_attritubes = ["short", "occ", "fast", "far", "easy"]
+
+font = cv2.FONT_HERSHEY_SIMPLEX
+fontScale = 1.2
+fontColor = (255, 255, 255)
+lineType = 3
+visualize = True
+check_track_label_folder = True
+
+
+def save_bev_img(
+    path_output_vis: str,
+    list_bboxes: List[Any],
+    list_difficulty_att: List[Any],
+    dataset_name: str,
+    log_id: str,
+    lidar_timestamp: int,
+    pc: np.ndarray,
+    egovehicle_to_city_se3: Any,
+) -> None:
+    """
+    Plot results on bev images and save 
+    """
+    image_size = 2000
+    image_scale = 10
+    img = np.zeros((image_size, image_size, 3))
+    pc = pc * image_scale
+    pc[:, 0] += int(image_size / 2)
+    pc[:, 1] += int(image_size / 2)
+    pc = pc.astype("int")
+
+    ind_valid = np.logical_and(
+        np.logical_and(pc[:, 0] >= 0, pc[:, 1] >= 0), np.logical_and(pc[:, 0] < image_size, pc[:, 1] < image_size)
+    )
+    img[pc[ind_valid, 0], pc[ind_valid, 1], :] = 0.4
+
+    path_imgs = os.path.join(path_output_vis, "bev")
+    if not os.path.exists(path_imgs):
+        os.mkdir(path_imgs)
+
+    for bbox, difficulty_att in zip(list_bboxes, list_difficulty_att):
+
+        qw, qx, qy, qz = bbox["rotation"]["w"], bbox["rotation"]["x"], bbox["rotation"]["y"], bbox["rotation"]["z"]
+        theta_local = np.arctan2(2 * (qw * qz + qx * qy), 1 - 2 * (qy * qy + qz * qz))
+        pose_local = np.array([bbox["center"]["x"], bbox["center"]["y"], bbox["center"]["z"]])
+
+        color_0 = 0.0
+        color_1 = 0.0
+        color_2 = 0.0
+        for att in difficulty_att:
+            color_0 += dict_color[att][0] / len(difficulty_att)
+            color_1 += dict_color[att][1] / len(difficulty_att)
+            color_2 += dict_color[att][2] / len(difficulty_att)
+        color = (color_0, color_1, color_2)
+
+        w, l, h = bbox["width"], bbox["length"], bbox["height"]
+        bbox_2d = np.array([[-l / 2, -w / 2, 0], [l / 2, -w / 2, 0], [-l / 2, w / 2, 0], [l / 2, w / 2, 0]])
+        R = np.array(
+            [[np.cos(theta_local), -np.sin(theta_local), 0], [np.sin(theta_local), np.cos(theta_local), 0], [0, 0, 1]]
+        )
+        bbox_2d = np.matmul(R, bbox_2d.transpose()).transpose() + pose_local[0:3]
```

Comment:
For the sake of simplicity, might be easier just to use the following snippet:
```
egovehicle_SE3_obj = SE3(rotation=quat2rotmat(self.quaternion), translation=self.translation)
bbox_ego_frame = egovehicle_SE3_obj .transform_point_cloud(bbox_object_frame)
```
Or alternatively we could just use the `as_2d_bbox()` function [here](https://github.com/argoai/argoverse-api/blob/master/argoverse/data_loading/object_label_record.py#L53). 
after `json_label_dict_to_obj_record(label)`

Reply:
I am not sure if using an SE object would increase the simplicity here.